src/components/Search/SearchHeader.tsx

```diff
@@ -67,13 +91,14 @@ export const SearchHeader = () => {
           query: removeEmpty({
             ...states[value as keyof typeof states],
             query: query || null, // Remove empty string from URL
+            page: 0, // restart the active page. TOOD: We need to consider whetehr or not it makes sense to store the page in the URL.
           }),
         },
         undefined,
         { shallow: true }
       );
     } else {
-      router.replace(`/search/${value}`);
+      router.replace(`/search/${value}?query=${query || ''}`, undefined, { shallow: true });
```

Comment:
Would it make sense to avoid adding the `query=` to the query string if query is empty?
```suggestion
      router.replace(`/search/${value}${query ? `?query=${query}` : ''}`, undefined, { shallow: true });
```